src/free-style.ts

```diff
@@ -169,12 +169,22 @@ function stringifyProperties (properties: Properties) {
   let result: string[] = []
 
   for (const [name, value] of properties) {
-    if (value != null) {
-      if (Array.isArray(value)) {
-        result.push(value.filter(x => x != null).map(x => styleToString(name, x)).join(';'))
-      } else {
-        result.push(styleToString(name, value))
-      }
+    // skip null values
+    if (value == null) {
+      continue
+    }
+    // resolve if a function
+    const value2 = typeof value === 'function' ? value() : value
+
+    // check again for null because value could be null
+    if (value2 == null) {
+      continue
+    }
```

Comment:
There's really no need for both these checks. Just moving the one check would have been adequate (two lines changed) as `typeof null === 'object'`.